src/move_mode.cpp

```diff
@@ -122,6 +122,22 @@ std::string move_mode::name() const
     return _name.translated();
 }
 
+std::string move_mode::type_name() const
+{
+    switch( _type ) {
+        case move_mode_type::PRONE:
+            return "prone";
+        case move_mode_type::CROUCHING:
+            return "crouching";
+        case move_mode_type::WALKING:
+            return "walking";
+        case move_mode_type::RUNNING:
+            return "running";
+        default:
+            return "???";
```

Comment:
This should probably raise an error or debug message instead of just returning "???"

Reply:
I decided to simply return "ERROR" - this should never happen, but it will be visible and easy to report when it does.